### PR TITLE
Create TriggersTemplate.gs

### PR DESCRIPTION
**Pull request by ria-arora**

### CODE REVIEW

File: TriggersTemplate.gs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * @param {Event} e The onSelectionChange event.
+ * e.authMode A value from the ScriptApp.AuthMode enum.
+ * e.source A Spreadsheet object, representing the Google Sheets file to which the script is bound.
+ * e.range A Range object, representing the cell or range of cells that were selected.
+ */
+function onSelectionChange(e) {
+}
+
+/**
+ * The event handler triggered when installing the add-on.
```
Comment (gargabhay) [inline TriggersTemplate.gs]: nit : "The event handler triggered when an add-on is installed."

File: TriggersTemplate.gs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * e.changeType The type of change 
+ * e.triggerUid ID of trigger that produced this event.
+ * e.user A User object, representing the active user, if available 
+ * @return {SpreadsheetTriggerBuilder} Builder for spreadsheet trigger chaining.
+ */
+function onChange(e) {
+}
+
+/**
+ * The event handler triggered when the selection changes in the spreadsheet.
```
Comment (gargabhay) [inline TriggersTemplate.gs]: I'd explain more on that. What does a selection mean? Some examples would be great. 
"The event handler triggered when selection changes in the spreadsheet. For example when a range is selected, when a different sheet is selected etc"

File: TriggersTemplate.gs [hunk 3]
```diff
@@ -0,0 +1,69 @@
+/**
+ * The event handler triggered when opening the spreadsheet.
+ * @param {Event} e The onOpen event.
```
Comment (gargabhay) [inline TriggersTemplate.gs]: nit: The onOpen event object. Here and everywhere else

File: TriggersTemplate.gs [hunk 4]
```diff
@@ -0,0 +1,69 @@
+/**
+ * The event handler triggered when opening the spreadsheet.
```
Comment (gargabhay) [inline TriggersTemplate.gs]: nit: when the attached spreadsheet is opened.

File: TriggersTemplate.gs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+ * e.source A Spreadsheet object, representing the Google Sheets file to which the script is bound.
+ * e.triggerUid ID of trigger that produced this event (installable triggers only).
+ * e.user User object representing the active user, if available.
+ * @return {SpreadsheetTriggerBuilder}  Builder for spreadsheet trigger chaining.
+ */
+function onOpen(e){
+}
+
+/**
+ * The event handler triggered when editing the spreadsheet.
```
Comment (gargabhay) [inline TriggersTemplate.gs]: nit: when an edit is made to the attached spreadsheet.

File: TriggersTemplate.gs [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+ * e.triggerUid ID of trigger that produced this event (installable triggers only).
+ * e.user A User object, representing the active user, if available.
+ * e.value New cell value after the edit. Only available if the edited range is a single cell. 
+ * @return {SpreadsheetTriggerBuilder} Builder for spreadsheet trigger chaining.
+ */
+function onEdit(e) {
+}
+
+/**
+ * The event handler triggered when the spreadsheet has a form submitted to it.
```
Comment (gargabhay) [inline TriggersTemplate.gs]: nit: when a response is submitted to the form linked to the attached spreadsheet.

File: TriggersTemplate.gs [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+ * @param {Event} e The onFormSubmit event.
+ * e.authMode A value from the ScriptApp.AuthMode enum.
+ * e.namedValues An object containing the question names and values from the form submission.
+ * e.range A Range object, representing the cell or range of cells that were edited.
+ * e.triggerUid ID of trigger that produced this event.
+ * e.values Array with values in the same order as they appear in the spreadsheet.
+ * @return {SpreadsheetTriggerBuilder} Builder for spreadsheet trigger chaining.
```
Comment (gargabhay) [inline TriggersTemplate.gs:36]: This trigger returns FormTriggerBuilder. Documentation : https://developers.google.com/apps-script/reference/script/form-trigger-builder#onFormSubmit()

File: TriggersTemplate.gs [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+ */
+function onFormSubmit(e) {
+}
+
+/**
+ * The event handler triggered when the spreadsheet's content or structure is changed.
```
Comment (gargabhay) [inline TriggersTemplate.gs]: nit: attached spreadsheet's content...

File: TriggersTemplate.gs [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+/**
+ * The event handler triggered when the selection changes in the spreadsheet.
+ * @param {Event} e The onSelectionChange event.
+ * e.authMode A value from the ScriptApp.AuthMode enum.
+ * e.source A Spreadsheet object, representing the Google Sheets file to which the script is bound.
+ * e.range A Range object, representing the cell or range of cells that were selected.
+ */
+function onSelectionChange(e) {
```
Comment (gargabhay) [inline TriggersTemplate.gs:59]: This also returns SpreadsheetTriggerBuilder

**Review by gargabhay — APPROVED** [review]
Thanks for the changes.

File: TriggersTemplate.gs [hunk 10]
```diff
@@ -0,0 +1,72 @@
+/**
+ * The event handler triggered when the attached spreadsheet is opened.
+ * @param {Event} e The onOpen event object.
+ * e.authMode A value from the ScriptApp.AuthMode enum.
+ * e.source A Spreadsheet object, representing the Google Sheets file to which the script is bound.
+ * e.triggerUid ID of trigger that produced this event (installable triggers only).
+ * e.user User object representing the active user, if available.
+ * @return {SpreadsheetTriggerBuilder}  Builder for spreadsheet trigger chaining.
```
Comment (paneendra-google) [inline TriggersTemplate.gs]: Is this for installable triggers only? Don't see it in the simple triggers

File: TriggersTemplate.gs [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+ */
+function onOpen(e){
+}
+
+/**
+ * The event handler triggered when an edit is made to the attached spreadsheet.
+ * @param {Event} e The onEdit event object.
+ * e.authMode A value from the ScriptApp.AuthMode enum.
```
Comment (paneendra-google) [inline TriggersTemplate.gs]: Indent this. and the lines below by 2 spaces to the right, also in other methods

File: TriggersTemplate.gs [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+ * @return {SpreadsheetTriggerBuilder} Builder for spreadsheet trigger chaining.
+ */
+function onFormSubmit(e) {
+}
+
+/**
+ * The event handler triggered when the attached spreadsheet's content or structure is changed.
+ * @param {Event} e The onChange event object.
+ * e.authMode A value from the ScriptApp.AuthMode enum.
+ * e.changeType The type of change
```
Comment (paneendra-google) [inline TriggersTemplate.gs]: Is there en enum that lists the type of changes? How will the user know the possible values for this field?

**Review by paneendra-google — CHANGES_REQUESTED** [review]

**Comment by ria-arora** [comment]
@googlebot I signed it!

**Review by paneendra-google — APPROVED** [review]

File: TriggersTemplate.gs (hunk 10 above)
Comment (ria-arora) [reply TriggersTemplate.gs]: Seems like it. Removed @return {SpreadsheetTriggerBuilder} comment for all the simple triggers.

File: TriggersTemplate.gs (hunk 12 above)
Comment (ria-arora) [reply TriggersTemplate.gs]: Added the types of changes.

**Pull request merged**